training/TRAINING.md

```diff
@@ -0,0 +1,213 @@
+```
+
+
+### 3. Create or find datasets
+
+To train your own Parler-TTS, you need datasets with 3 main features:
+- speech data
+- text transcription of the speech data
+- conditionning text description - that you can create using [Data-Speech](https://github.com/huggingface/dataspeech), a library that allows you to annotate the speaker and utterance characteristics with natural language description.
+
+Note that we made the choice to use description of the main speech characteristics (speaker pitch, speaking rate, level of noise, etc.) but that you are free to use any handmade or generated text description that makes sense.
+
+In the rest of this guide, and to make it simple, we'll use the [4.8K-samples clean test split](https://huggingface.co/datasets/blabble-io/libritts_r/viewer/clean/test.clean) of [LibriTTS-R](https://huggingface.co/datasets/blabble-io/libritts_r/). We've annotated LibriTTS-R using [Data-Speech](https://github.com/huggingface/dataspeech) and shared the resulting dataset here: [parler-tts/libritts_r_tags_tagged_10k_generated](https://huggingface.co/datasets/parler-tts/libritts_r_tags_tagged_10k_generated).
+
+
+## C. Training
+
+The script [`run_parler_tts_training.py`](/training/run_parler_tts_training.py) is an end-to-end script that:
+1. load dataset(s) and merge them to the annotation dataset(s) if necessary
+2. pre-compute audio tokens
+3. train Parler-TTS
+
+In this example, we will train and evaluate on a subsample of the test split. This is purely to demonstrate how to use the training script, rather than recommended advice for defining train/validation splits. We advise that you train on the train splits of your dataset, evaluate and tune hyper-parameters on the validation split, and only test the final checkpoint on the test split.
+
+```sh
+accelerate launch ./training/run_parler_tts_training.py \
+    --model_name_or_path "./parler-tts-untrained-300M/parler-tts-untrained-300M/" \
+    --feature_extractor_name "parler-tts/dac_44khZ_8kbps" \
+    --description_tokenizer_name "google/flan-t5-base" \
+    --prompt_tokenizer_name "google/flan-t5-base" \
+    --report_to "wandb" \
+    --overwrite_output_dir true \
+    --train_dataset_name "blabble-io/libritts_r" \
+    --train_metadata_dataset_name "parler-tts/libritts_r_tags_tagged_10k_generated" \
+    --train_dataset_config_name "clean" \
+    --train_split_name "test.clean" \
+    --eval_dataset_name "blabble-io/libritts_r" \
+    --eval_metadata_dataset_name "parler-tts/libritts_r_tags_tagged_10k_generated" \
+    --eval_dataset_config_name "clean" \
+    --eval_split_name "test.clean" \
+    --target_audio_column_name "audio" \
+    --description_column_name "text_description" \
+    --prompt_column_name "text" \
+    --max_duration_in_seconds 20 \
+    --min_duration_in_seconds 2.0 \
+    --add_audio_samples_to_wandb true \
+    --id_column_name "id" \
+    --preprocessing_num_workers 8 \
+    --pad_token_id 1024 \
+    --decoder_start_token_id 1025 \
```

Comment:
Decoder start and pad token ids could be set in the model init script right? They seem more relevant there no?